### PR TITLE
fix: added a Parse function for the command-line arguments

### PR DESCRIPTION
**Pull request by alexal**
This is a potential fix for #1034  . Seems like that we have to run a Parse() function, at least it started working in my local copy. @wtrocki , @craicoverflow what do you think?

### CODE REVIEW

**Comment by wtrocki** [comment]
Issue number if wrong :)

**Comment by alexal** [comment]
> Issue number if wrong :)

Right, the correct one is #1034

File: pkg/cmd/root/root.go [hunk 1]
```diff
@@ -41,6 +42,7 @@ func NewRootCommand(f *factory.Factory, version string) *cobra.Command {
 	cmd.Version = version
 
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	_ = fs.Parse(os.Args[1:])
```
Comment (wtrocki) [inline pkg/cmd/root/root.go:45]: What this does?

**Comment by wtrocki** [comment]
I cannot reproduce this:

- rhoas status  -v (works)
- rhoas kafka list  -v (works)
- rhoas service-registry artifact list -v (works)

The only command that doesn't work is login.

File: pkg/cmd/root/root.go (hunk 1 above)
Comment (alexal) [reply pkg/cmd/root/root.go:45]: Parses a command line into the defined flags.

**Comment by alexal** [comment]
> The only command that doesn't work is login.
I see, let me take a look.

**Comment by wtrocki** [comment]
No need to close. I thinkl this is the reason: https://github.com/redhat-developer/app-services-cli/blob/main/pkg/cmd/debug/flag.go#L8:L8

**Comment by wtrocki** [comment]
SOrry wrong link. I meant this: https://github.com/redhat-developer/app-services-cli/blob/main/pkg/cmd/factory/default.go#L34:L34 returning false for login but ok for the rest

**Comment by alexal** [comment]
> https://github.com/redhat-developer/app-services-cli/blob/main/pkg/cmd/factory/default.go#L34:L34
What is interesting, it is also false for a status command. So, most likely it was changed somewhere later in the stack.
image

**Comment by wtrocki** [comment]
Yes.. it is the same. It is never enabled appently. Very strange. My take is that we should not enable this in factory, but later after flag is setup and can be detected.

**Comment by alexal** [comment]
This piece of code

c.logger.Debug("Refreshing tokens")

Gets a logger from KeycloakConnection, not from the status command. And for KeycloakConnection logger, debugEnabled is set to true:

image

**Comment by wtrocki** [comment]
That is it. you cracked system

**Comment by alexal** [comment]
@wtrocki ,

I know where is the problem. This line in default.go

debugEnabled := debug.Enabled()

is called before these lines in root.go

arguments.AddDebugFlag(fs)
...
pflag.CommandLine.AddGoFlagSet(flag.CommandLine)

This is why debugEnabled will always be set to false.

**Comment by wtrocki** [comment]
exactly. Moving this after will fix jt

**Pull request closed without merging**